homeassistant/components/weatherflow_cloud/weather.py

```diff
@@ -0,0 +1,241 @@
+"""Support for WeatherFlow Forecast weather service."""
+from __future__ import annotations
+
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any
+
+from homeassistant.components.weather import (
+    DOMAIN as WEATHER_DOMAIN,
+    Forecast,
+    SingleCoordinatorWeatherEntity,
+    WeatherEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_NAME,
+    UnitOfPrecipitationDepth,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.dt import utc_from_timestamp
+from homeassistant.util.unit_system import METRIC_SYSTEM
+
+from .const import (
+    ATTR_ATTRIBUTION,
+    CONF_FIRMWARE_REVISION,
+    CONF_STATION_ID,
+    DEFAULT_NAME,
+    DOMAIN,
+    MANUFACTURER,
+    MODEL,
+)
+from .coordinator import WeatherFlowCloudDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a weather entity from a config_entry."""
+    coordinator: WeatherFlowCloudDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    entity_registry = er.async_get(hass)
+
+    name: str | None
+    is_metric = hass.config.units is METRIC_SYSTEM
+
+    if (name := config_entry.data.get(CONF_NAME)) and name is None:
+        name = DEFAULT_NAME
+    elif TYPE_CHECKING:
+        assert isinstance(name, str)
```

Comment:
We don't even store a name

Reply:
In the current code it is storing a name. Although I went back and forth on this:

<img width="545" alt="image" src="https://github.com/home-assistant/core/assets/6491743/f9010327-8aa5-4a49-a684-084e8bf6c911">

At a minimum we could just store the credentials and at startup figure out which stations to add...

Reply:
Oh lol, I forgot this was a thing. But I actually prefer to store as less as possible. This way if the station name changes HA can pick it up without updating the config entry

Reply:
I'm hoping we can just store a SINGLE api token which would be awesome.